### PR TITLE
[SQL] Move SLT into a separate project

### PR DESCRIPTION
**Pull request by mihaibudiu**
Is this a user-visible change (yes/no): no

A reimplementation of #943

### CODE REVIEW

File: sql-to-dbsp-compiler/README.md [hunk 1]
```diff
@@ -90,7 +90,7 @@ See the [documentation](https://www.feldera.com/docs/sql/intro)
 
 ## Compiler architecture
 
-[A presentation](https://github.com/feldera/feldera/blob/main/docs/static/sql-compiler-architecture.pptx)
+[A presentation](https://mihaibudiu.github.io/work/sql-compiler-architecture23.pptx)
```
Comment (lalithsuresh) [inline sql-to-dbsp-compiler/README.md:93]: Would be better to point to the feldera URL.

**Review by lalithsuresh — APPROVED** [review]

**Review by lalithsuresh — COMMENTED** [review]
If you're moving these files around anyway, might as well turn SQL-compiler to sql-compiler :)

File: sql-to-dbsp-compiler/README.md (hunk 1 above)
Comment (mihaibudiu) [reply sql-to-dbsp-compiler/README.md:93]: there is no feldera URL, I don't know where this file has gone.

**Comment by mihaibudiu** [comment]
> If you're moving these files around anyway, might as well turn SQL-compiler to sql-compiler :)

I am afraid to do it, let's do it some other time

**Pull request merged**